mscore/plugin/api/selection.h

```diff
@@ -0,0 +1,55 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2019 Werner Schweer and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2
+//  as published by the Free Software Foundation and appearing in
+//  the file LICENCE.GPL
+//=============================================================================
+
+#ifndef __PLUGIN_API_SELECTION_H__
+#define __PLUGIN_API_SELECTION_H__
+
+#include "elements.h"
+#include "score.h"
+
+namespace Ms {
+namespace PluginAPI {
+
+//---------------------------------------------------------
+//   Selection
+//    Wrapper class for internal Ms::Selection
+///  \since MuseScore 3.3 
+//---------------------------------------------------------
+
+class Selection : public QObject {
+      Q_OBJECT
+      /// Current GUI selections for the score.
+      /// \since MuseScore 3.3
+      Q_PROPERTY(QQmlListProperty<Ms::PluginAPI::Element> elements READ elements)
+
+      /// \cond MS_INTERNAL
+
+   protected:
+      Ms::Selection* _select;
+      Score*         _score;
```

Comment:
It is probably better not to store a pointer to a `Score` wrapper in this object. That wrapper can be deleted at any time when QML engine decides it is not necessary anymore. If you intended to use it as an argument to `wrapContainerProperty`, it is probably better just to use `this` as an argument, this argument is then passed to the [`QQmlListProperty` constructor](https://doc.qt.io/qt-5/qqmllistproperty.html#QQmlListProperty-3) as the first argument.

Reply:
I believe this is a remnant of my trial to get it working.
-Dale